app/assets/javascripts/discourse/app/models/post-stream.js

```diff
@@ -617,26 +617,33 @@ export default RestModel.extend({
     }
 
     const loadedAllPosts = this.loadedAllPosts;
+    this._loadingPostIds = this._loadingPostIds || [];
 
     if (this.stream.indexOf(postId) === -1) {
-      this.stream.addObject(postId);
       if (loadedAllPosts) {
+        if (this._loadingPostIds.indexOf(postId) === -1) {
+          this._loadingPostIds.push(postId);
+        }
         this.set("loadingLastPost", true);
-        return this.findPostsByIds([postId])
+        return this.findPostsByIds(this._loadingPostIds, opts)
           .then((posts) => {
+            this._loadingPostIds = undefined;
```

Comment:
It can seem like a nitpick, but I generally prefer to use null in this case, undefined means it has never been assigned, which is not the case here.

```suggestion
            this._loadingPostIds = null;
```